packages/grid/x-data-grid-pro/src/components/GridRowReorderCell.tsx

```diff
@@ -0,0 +1,101 @@
+import * as React from 'react';
+import { unstable_composeClasses as composeClasses } from '@mui/material';
+import {
+  GridEvents,
+  GridRenderCellParams,
+  GridRowEventLookup,
+  gridRowTreeDepthSelector,
+  gridSortModelSelector,
+  useGridApiContext,
+  useGridSelector,
+  getDataGridUtilityClass,
+  gridEditRowsStateSelector,
+} from '@mui/x-data-grid';
+import { isObjectEmpty } from '@mui/x-data-grid/utils/utils';
+import { DataGridProProcessedProps } from '../models/dataGridProProps';
+import { useGridRootProps } from '../hooks/utils/useGridRootProps';
+
+type OwnerState = {
+  classes?: DataGridProProcessedProps['classes'];
+  isDraggable: boolean;
+};
+
+const useUtilityClasses = (ownerState: OwnerState) => {
+  const { isDraggable, classes } = ownerState;
+
+  const slots = {
+    root: ['rowReorderCell', isDraggable && 'rowReorderCell--draggable'],
+    container: ['rowDraggableContainer'],
+  };
+
+  return composeClasses(slots, getDataGridUtilityClass, classes);
+};
+
+const GridRowReorderCell = (params: GridRenderCellParams) => {
+  const apiRef = useGridApiContext();
+  const rootProps = useGridRootProps();
+  const sortModel = useGridSelector(apiRef, gridSortModelSelector);
+  const treeDepth = useGridSelector(apiRef, gridRowTreeDepthSelector);
+  const editRowsState = useGridSelector(apiRef, gridEditRowsStateSelector);
+  // eslint-disable-next-line no-underscore-dangle
+  const cellValue = params.row.__reorder__ || params.row.id;
```

Comment:
`params.row.id` won't work if another field is used as ID. You can replace it with `params.id`.